gobblin-hive-registration/src/test/java/org/apache/gobblin/hive/metastore/HiveMetaStoreUtilsTest.java

```diff
@@ -212,4 +217,84 @@ public void testGetHiveTable() throws Exception {
     Assert.assertEquals(fieldA.getType(), "int");
 
   }
+
+  @Test
+  public void testContainsUnionType_AvroSucceeds() {
+    final String databaseName = "db";
+    final String tableName = "tbl";
+    final State serdeProps = new State();
+
+    serdeProps.setProp("avro.schema.literal", "{\"type\": \"record\", \"name\": \"TestEvent\","
+        + " \"namespace\": \"test.namespace\", \"fields\": [{\"name\":\"aUnionField\", \"type\": [\"string\", \"int\"]}]}");
+    HiveTable hiveTable = createTestHiveTable_Avro(databaseName, tableName, serdeProps);
+
+    Assert.assertTrue(HiveMetaStoreUtils.containsUnionTypeColumn(hiveTable));
```

Comment:
in the function above it looks like its looking for a `uniontype` column name and that's not here although i do see `aUnionField` has multiple options for type.

Reply:
non optional uniontypes in avro are represented as a ["type1", "type2", ...] where the first type is the default. See https://avro.apache.org/docs/1.10.2/spec.html#Unions

Reply:
And so when we register a table with this schema to hive, the column type will contain "uniontype". Hence why this test passes.